From c32727c6a173b118479cfd00ec93b6540bae6039 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 17 Jun 2019 19:47:35 +0300
Subject: [PATCH] [SW-0] goya: fix uart/spi baud rate calculation

In Goya (unlike in Gaudi), the UART and SPI clock is derived from the
PCI PLL divisor output 1, and not output 2 (as is currently).
This worked till now because both outputs were the same (reference
clock) and thus the calculation produced the correct values.
Correct the source for both UART and SPI baud rate calculations.

Change-Id: I725b89df7e0e765ed3625c9b08cc2f9231355a3a
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/goya/asic.c b/goya/asic.c
index 54a183b..4a4f510 100644
--- a/goya/asic.c
+++ b/goya/asic.c
@@ -66,17 +66,17 @@
 
 void hl_uart_init(void)
 {
-	u32 pcie_div2_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV2);
+	u32 uart_clk_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV1);
 
-	preboot_uart_init(pcie_div2_freq / HL_UART_BAUDRATE);
+	preboot_uart_init(uart_clk_freq / HL_UART_BAUDRATE);
 }
 
 u32 hl_spi_baudr_get(void)
 {
 	u32 factor;
-	u32 pcie_div2_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV2);
+	u32 spi_clk_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV1);
 
-	factor = pcie_div2_freq / HL_SPI_FLASH_MAX_FREQ;
+	factor = spi_clk_freq / HL_SPI_FLASH_MAX_FREQ;
 
 	return factor > 2 ? factor : 2;
 }
